src/kudu/consensus/raft_consensus.cc

```diff
@@ -3443,12 +3453,16 @@ void RaftConsensus::CompleteConfigChangeRoundUnlocked(ConsensusRound* round, con
 
 void RaftConsensus::EnableFailureDetector(boost::optional<MonoDelta> delta) {
   if (PREDICT_TRUE(FLAGS_enable_leader_failure_detection)) {
+    persistent_vars_->enable_failure_detector();
+    CHECK_OK(persistent_vars_->Flush());
     failure_detector_->Start(std::move(delta));
   }
 }
 
 void RaftConsensus::DisableFailureDetector() {
```

Comment:
What we really need is to not start election (when this special flag is set). So, maybe gate starting election in RaftConsensus::StartElection() based on the new flag (implemented here or in plugin as being discussed elsewhere). This has the added benefit of handling all kind of elections/promotions and avoids race conditions (for example, a pre-election could have already been running when the flag was flipped which might be followed by a real election soon)

Reply:
This is a great point, Vinay. Let's get rid of the enable/disable failure detector stuff and just no-op elections while a persistent var such as "allow_start_elections" is set to false (we should default it to true if unset)